pkg/rtc/subscriptionmanager.go

```diff
@@ -449,6 +449,8 @@ func (m *SubscriptionManager) subscribe(s *trackSubscription) error {
 		return ErrSubscriptionLimitExceeded
 	}
 
+	s.setPublisher(res.PublisherIdentity, res.PublisherID)
+
```

Comment:
Moved this up here from below as there is a subscription permission update below and that was sending `publisherID` empty on the first permission update.